### PR TITLE
Create JAX Advanced Tutorials (section, move/create docs)

### PR DESCRIPTION
**Pull request by 8bitmp3**
@jakevdp

### CODE REVIEW

File: docs/advanced_tutorials.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.. _advanced-tutorials:
```
Comment (jakevdp) [inline docs/advanced_tutorials.rst:1]: Can we make this a section in `tutorials.rst` rather than creating another file?

**Review by jakevdp — CHANGES_REQUESTED** [review]
Several of these sections are duplicative of existing docs (e.g. `docs/external-callbacks.md` and `docs/notebooks/external_callbacks.md`.

More work is necessary to de-duplicate these, update indices, and re-direct old URLs to the new locations. See #21064 for an example of how we did this in the past

Also, rather than creating new files, we should move the old files out of `docs/_tutorials` to their new location, in order to preserve change history.

File: docs/advanced_tutorials.rst (hunk 1 above)
Comment (8bitmp3) [reply docs/advanced_tutorials.rst]: Done!

**Review by jakevdp — COMMENTED** [review]
Errors are due to the change noted below; also it looks like there are two remaining uses of the `understanding-jaxprs` reference that need to be updated, one in `glossary.rst` and one in `jit-compilation.md`.

File: docs/tutorials.rst [hunk 2]
```diff
@@ -16,3 +17,13 @@ Tutorials
...
+   
+.. toctree::
+   :maxdepth: 1
+   :caption: Advanced tutorials
+
+   advanced-autodiff
+   external-callbacks
+   gradient-checkpointing
+   jax-primitives
+   jaxpr_language
```
[the remark this answers is not in the document]
Comment (8bitmp3) [reply docs/tutorials.rst:29]: Good catch! Done

File: docs/conf.py [hunk 3]
```diff
@@ -85,7 +85,6 @@ def _do_not_evaluate_in_jax(
 ]
 
 intersphinx_mapping = {
-    'array_api': ('https://data-apis.org/array-api/2023.12/', None),
```
Comment (jakevdp) [inline docs/conf.py:88]: please revert

Comment (8bitmp3) [reply docs/conf.py:88]: That's a newly added line. Will do!!

**Comment by 8bitmp3** [comment]
@jakevdp PTAL

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
As we discussed, the structure of the left-nav dropdown is still not exactly how we'd like it, but we can land and iterate. Thanks for working on this!

**Review by ayaka14732 — APPROVED** [review]

**Pull request merged**